android/app/src/androidTest/java/org/ZingoLabs/Zingo/RestoreFromUfvk.kt

```diff
@@ -0,0 +1,48 @@
+package org.ZingoLabs.Zingo
+
+import java.io.File
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.experimental.categories.Category
+
+@Category(IntegrationTest::class)
+class RestoreFromUfvkTest {
+
+    data class Addresses (
+    	val address : String,
+    	val receivers : Receivers
+    )
+
+    data class Receivers (
+    	val transparent : String,
+    	val sapling : String,
+    	val orchard_exists : Boolean
+    )
+
+    @Test
+    fun restoreFromUfvkTest() {
+        // Fail if wallet already exists
+        val walletFile = File(MainApplication.getAppContext()?.filesDir, "wallet.dat")
+        if (walletFile.exists()) {
+            assertThat(true).isEqualTo(false)
+        }
+        
+        val server = "https://mainnet.lightwalletd.com:9067"
+        val chainhint = "main"
+        val ufvk = "uview1vmle9qzgsmzn8qskg06cjjt406eq3euvkn8un3t6sk7fjd72qp72guyskktr2gvgcpmd99qgquxdw7s54v0dfxn3degzhxp9gmvpve6vlvv459pxfre05v6l47aqx47rgh86t6n77svstd8ff6d8cwtn98uq66k6u5jqrpyz0pqflkppfq3djscrd2acnc7ymkd6fssk0t4rh2ynaux7z4ylt38jqgjhnu88h2jz8qwd3t5dwsc3ycvsea6grs4zg76r9vw48r9zvlphnpfsgc5eqpum7hghdm2eguw4h6n9m8rhuzh0qdc24z4z5ftcv6xxvvs3yrugea23xlys3f9qv0fenh0xp0hej8enlr82esl54hn27d6432kygwqx700ez84e72f03vtmece73dftpjvv3v7w65yaz2rjwmarxjzrnn02u5kx4p7a42k5lxgcgqgwjyp9w6x24ccm0dvlf4637ss6l3xmuv2strza60c5k5uasdcn"
+        val birthday = "419200"
+        val datadir = MainApplication.getAppContext()!!.filesDir.path
+
+        // RustFFI.initfromufvk(server, ufvk, birthday, datadir, chainhint)
```

Comment:
If you are using this line... you need to know what is the response of this function, normally will be this: `Error: This wallet is watch-only.`, in this case, everything is OK, but here could be different result.

Reply:
seems strange that the happy path returns an "Error: ... " ?